### ISSUE TITLE
Add isn't starting after update: Running as root is not recommended

### ISSUE THREAD

**Issue by senfkorn** [hassio-addons/addon-jupyterlab#319]
Hi there,

After the last update yesterday, my jupyterlab addon is unable to start. 

Here is a short part of the log:

[15:43:01] INFO: Starting the Jupyter server...
[I 2022-08-22 15:43:12.807 ServerApp] jupyterlab | extension was successfully linked.
[I 2022-08-22 15:43:12.850 ServerApp] nbclassic | extension was successfully linked.
[I 2022-08-22 15:43:17.985 ServerApp] notebook_shim | extension was successfully linked.
[I 2022-08-22 15:43:18.166 ServerApp] notebook_shim | extension was successfully loaded.
[I 2022-08-22 15:43:18.182 LabApp] JupyterLab extension loaded from /usr/local/lib/python3.9/dist-packages/jupyterlab
[I 2022-08-22 15:43:18.182 LabApp] JupyterLab application directory is /usr/local/share/jupyter/lab
[I 2022-08-22 15:43:18.190 ServerApp] jupyterlab | extension was successfully loaded.
[I 2022-08-22 15:43:18.221 ServerApp] nbclassic | extension was successfully loaded.
[C 2022-08-22 15:43:18.223 ServerApp] Running as root is not recommended. Use --allow-root to bypass.
[15:43:18] WARNING: Jupyter crashed, halting add-on
[15:43:18] INFO: Jupyter stopped, restarting...
s6-rc: info: service legacy-services: stopping
[15:43:19] INFO: NGINX stopped, restarting...
s6-svwait: fatal: supervisor died
s6-rc: info: service legacy-services successfully stopped
s6-rc: info: service legacy-cont-init: stopping
s6-rc: info: service legacy-cont-init successfully stopped
s6-rc: info: service fix-attrs: stopping
s6-rc: info: service fix-attrs successfully stopped
[15:43:19] INFO: Jupyter stopped, restarting...
s6-rc: info: service s6rc-oneshot-runner: stopping
s6-rc: info: service s6rc-oneshot-runner successfully stopped

I've tried adding --allow-root to the init_commands but this dosn't work.
Any other ideas?

br
Andy

**Comment by frenck** [comment]
WTH did it run in my pre-release testing?! 😭 

Will take a look!


../Frenck

**Comment by senfkorn** [comment]
The following Version is running:

 Add-on version: 0.8.0
 There is an update available for this add-on!
 Latest add-on version: 0.8.1
 Please consider upgrading as soon as possible.
 System: Debian GNU/Linux 11 (bullseye)  (amd64 / qemux86-64)
 Home Assistant Core: 2022.8.6
 Home Assistant Supervisor: 2022.08.3


I've downgraded to this version and its running again

**Comment by atmezferix** [comment]
Crashes the same for me.  I rolled back to 0.8.0 and working again.  Same logs as about.

**Comment by hommadi2001** [comment]
Getting the same issue and crash logs with v0.8.1.
Dont know how to downgrade an ad-on, and don't have any prev. backups (n00b here).

**Comment by BugMar1** [comment]
Hi, same problem here, has anyone found a solution?

**Comment by kruzhkov** [comment]
Hi there,

Getting the same issue and crash logs with v0.8.1.

**Comment by QcRaphRoseQc** [comment]
Same issue with 8.1 how can I downgrade to v8.0 in homeAssistant ?

**Comment by xanderificnl** [comment]
Sent a PR (#321)—which should resolve this issue. I'm guessing just setting the config option isn't enough any more.

Anyhow, for those knowledgeable enough—i.e. **don't do this unless you're a jackal that likes to live dangerously**:

```
cp /mnt/data/docker/overlay2/42a874b93a0a31ad89d7d87291f51820c9cba340521507992b9b90516ac24fd4/diff/etc/services.d/jupyter/run{,.orig}
sed -i 's/exec jupyter lab/exec jupyter lab --allow-root/g' /mnt/data/docker/overlay2/42a874b93a0a31ad89d7d87291f51820c9cba340521507992b9b90516ac24fd4/diff/etc/services.d/jupyter/run
# start jupyter
```

**Comment by mvaneijken** [comment]
Until the fix is released, I used the following steps to run the 0.7.0 version (based on this [article](https://community.home-assistant.io/t/how-to-downgrade-any-supervisor-addon/349649)):

1. Stop the Jupyterlab add-on and disable autostart/watchdog
2. Open a terminal
3. Download the 0.7.0 version: `sudo docker pull ghcr.io/hassio-addons/jupyterlab/amd64:0.7.0`
4. Delete the 0.8.1 version: `sudo docker rmi ghcr.io/hassio-addons/jupyterlab/amd64:0.8.1`
6. Re-tag version 0.7.0 with tag 0.8.1: `sudo docker tag ghcr.io/hassio-addons/jupyterlab/amd64:0.7.0 ghcr.io/hassio-addons/jupyterlab/amd64:0.8.1`
7. Start the Jupyterlab add-on .
